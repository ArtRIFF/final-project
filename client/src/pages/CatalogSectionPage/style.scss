@import '../../global_scss/mixins.scss';
@import '../../global_scss/vars.scss';

.grid-wrapper {
   padding: 30px 0 95px;
   display: grid;
   grid-gap: 10px;
   grid-template-areas:
      "breadcrumbs breadcrumbs"
      "asideFilter catalogPageImg"
      "asideFilter filter"
      "asideFilter categoryCards"
      "asideFilter paginnation";

      @media (max-width:$desktop-size) {
         grid-gap: 0px;
      }
}

.catalogPageImg-wrapper {
   grid-area: catalogPageImg;
   width: 100%;
   min-height: 100%;
   
   img {
      display: inline-block;
      width: inherit;
      height: inherit;
   }
}

.breadcrumbs__catalog {
   margin-top: 30px;
   margin-left: 10px;
}

.asideFilter-wrapper {
   @media (max-width:$desktop-size) {
      width: 0;
      overflow: hidden;
   }



   grid-area: asideFilter;
   width: 300px;
   min-height: 508px;
   position: relative;
   width: '300px'

   &--show {
      width: 200px;
   }
}

.asideFilter-wrapper--show .aside-filter {
   background-color: $additional-text-color;
   border-radius: 4px;
   animation: slide 0.8s ease-in-out;
   padding: 10px;
}


@keyframes slide {
   0% {
      left: -100%;
   }

   100% {
      left: 0%;
   }
}

.filter-wrapper {
   grid-area: filter;
   width: 100%;
   min-height: 43px;
   padding: 15px 5px;
}

.categoryCards-wrapper {
   min-height: 350px;

   @media (max-width: 1196px) {
      min-height: 580px;
   }

   @media (max-width: $desktop-size) {
      min-height: 580px;
   }

   grid-area: categoryCards;
   overflow: hidden;
}


.paginnation-wrapper {
   grid-area: paginnation;
}