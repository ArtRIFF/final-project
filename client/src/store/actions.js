import { createAction } from "@reduxjs/toolkit";
import { getCards, getFilteredCards } from "../API/cardsAPI";

export const setModalRender = createAction("SET_MODAL_RENDER");
export const setAllCollectionProduct = createAction("SET_ALL_COLLECTION_PRODUCT");
export const setFilteredProducts = createAction("SET_FILTERED_PRODUCTS");
export const setNewCollectionProduct = createAction("SET_NEW_COLLECTION_PRODUCT");
export const setBestsellers = createAction("SET_BESTSELLERS");
export const setOutlet = createAction("SET_OUTLET");
export const setInCart = createAction("SET_IN_CART");
export const changeCart = createAction("CHANGE_CART");
export const setInFavorite = createAction("SET_IN_FAVORITE");
export const removeFromFavorite = createAction("REMOVE_FROM_IN_FEVORITE");

export const fetchAllCollectionProduct = () => (dispatch) => {
  return getCards().then((data) => {
    dispatch(setAllCollectionProduct(data));
  });
};

export const fetchFilteredProducts = (stringFilteredParam) => (dispatch) => {
  return getFilteredCards(stringFilteredParam).then((data) => {
    dispatch(setFilteredProducts(data));
  });
};

// export const fetchNewCollectionProduct = () => (dispatch) => {
//   return getCards().then((data) => {
//     dispatch(
//       setNewCollectionProduct(
//         data.filter((element) => element.statusProduct === "NEW")
//       )
//     );
//   });
// };

// export const fetchBestsellers = () => (dispatch) => {
//   return getCards().then((data) => {
//     dispatch(
//       setBestsellers(
//         data.filter((element) => element.statusProduct === "BESTSELLER")
//       )
//     );
//   });
// };

// // export const fetchOutlet = () => (dispatch) => {
// //   return getCards().then((data) => {
// //     dispatch(
// //       setOutlet(data.filter((element) => element.statusProduct === "OUTLET"))
// //     );
// //   });
// // };
