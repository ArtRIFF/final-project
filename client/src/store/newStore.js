import { configureStore } from "@reduxjs/toolkit";
import { applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import thunk from "redux-thunk";
import logger from "redux-logger";
import productsSlice from "./products/productsSlice";
import favoriteSlice from "./favorite/favoriteSlice";
import cartSlice from "./cart/cartSlice";
import filteredProductsSlice from "./filteredProducts/filteredProductsSlice"

const store = configureStore ({
    reducer: {
        products: productsSlice.reducer,
        cart: cartSlice.reducer,
        favorite: favoriteSlice.reducer,
        filteredProducts: filteredProductsSlice.reducer,
        // composeWithDevTools(applyMiddleware(thunk, logger))
    },
    // middleware: 
})

export default store
