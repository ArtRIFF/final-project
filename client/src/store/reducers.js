// // import { createReducer } from "@reduxjs/toolkit";
// // import * as actions from "./actions";

// // const localStItemsFavorite = localStorage.getItem("inFavorite")
// //   ? JSON.parse(localStorage.getItem("inFavorite"))
// //   : [];
// // const localStItemsCart = localStorage.getItem("inCart")
// //   ? JSON.parse(localStorage.getItem("inCart"))
// //   : [];

// export const defaultState = {
//   modalRender: false,
//   newCollectionProduct: [],
//   allCollectionProduct: [],
//   filteredProducts: [],
//   bestsellers: [],
//   outlet: [],
//   inCart: localStItemsCart,
//   inFavorite: localStItemsFavorite,
// };

// export default createReducer(defaultState, {
//   [actions.setModalRender]: (state, { payload }) => {
//     state.modalRender = payload;
//   },
//   [actions.setAllCollectionProduct]: (state, { payload }) => {
//     state.allCollectionProduct = payload;
//   },
//   [actions.setFilteredProducts]: (state, { payload }) => {
//     state.filteredProducts = payload;
//   },
//   [actions.setNewCollectionProduct]: (state, { payload }) => {
//     state.newCollectionProduct = payload;
//   },

//   [actions.setBestsellers]: (state, { payload }) => {
//     state.bestsellers = payload;
//   },
//   [actions.setOutlet]: (state, { payload }) => {
//     state.outlet = payload;
//   },
//   [actions.setInCart]: (state, { payload }) => {
//     state.inCart = [...state.inCart, payload];
//   },
//   [actions.changeCart]: (state, { payload }) => {
//     state.inCart = payload;
//   },
//   [actions.setInFavorite]: (state, { payload }) => {
//     state.inFavorite = [...state.inFavorite, payload];
//   },
//   [actions.removeFromFavorite]: (state, { payload }) => {
//     state.inFavorite = payload;
//   }
// });
