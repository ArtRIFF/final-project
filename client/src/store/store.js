import { createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import thunk from "redux-thunk";
import logger from "redux-logger";

import rootReducers from "./redusers";

const store = createStore(
  rootReducers,
  composeWithDevTools(applyMiddleware(thunk, logger))
);

export default store;
