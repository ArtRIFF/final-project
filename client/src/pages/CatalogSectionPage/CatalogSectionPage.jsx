import "./style.scss";
import Breadcrumbs from "./components/Breadcrumbs/Breadcrumbs";
import CategoryFilter from "./components/CategoryFilter/CategoryFilter";
import AsideFilter from "./components/AsideFilter/AsideFilter";
import Items from "./components/Pagination/Items";
import Pagination from "./components/Pagination/Pagination";

import { useDispatch, useSelector } from "react-redux";
import { useState, useEffect } from "react";

import { selectorAllCollectionProduct } from "../../store/selectors";
import { fetchAllCollectionProduct } from "../../store/actions";

const CatalogSectionPage = ({ alreadyFilteredArray }) => {
  const dispatch = useDispatch();

  const [items, setitems] = useState([
    11, 124, 1244, 1241, 12, 12, 1, 12, 12, 12, 3, 4, 5, 6, 7, 3, 12, 12, 12,
    124, 15, 12, 53735, 12312,
  ]);
  const [loading, setLoading] = useState(false);
  const [currentPage, setCurrentPage] = useState(1);
  const [itemsPerPage] = useState(5);
  const [filtredArray, setFiltredArray] = useState(alreadyFilteredArray);
  const lastItemIndex = currentPage * itemsPerPage;
  const firstItemIndex = lastItemIndex - itemsPerPage;
  const currentItem = items.slice(firstItemIndex, lastItemIndex);

  useEffect(() => {
    const getItems = async () => {
      setLoading(true);
      // const res = await axios.get("https://final-backend-new.onrender.com"); //адреса сторінки на бекенді для отримання елементів сторінки
      // setitems(res.data);
      setLoading(false);
    };
    getItems();
  }, []);
  const [showAsideFilter, setModalRender] = useState(false);
  const allCollectionArray = useSelector(selectorAllCollectionProduct);

  const callAsideFilter = () => {
    setModalRender(true);
  };

  useEffect(() => {
    dispatch(fetchAllCollectionProduct());
  }, []);

  const filterRequest = (array) => {
    setFiltredArray(array);
  };

  const hideAsideFilter = (event) => {
    const isFilterElement = !!event.target.closest(
      ".asideFilter-wrapper--show"
    );
    const isCallButton = !!event.target.closest(".category-filter--btn");

    if (event && !isFilterElement && !isCallButton) {
      setModalRender(false);
    }
  };

  const array = Array.isArray(filtredArray)
    ? filtredArray.length
    : allCollectionArray.length;

  return (
    <div className="container" onClick={hideAsideFilter}>
      <div className="grid-wrapper">
        <div className="breadcrumbs-wrapper">
          <Breadcrumbs />
        </div>
        <div className="catalogPageImg-wrapper">
          <img
            src="img/catalogSectionPage/CategorySectionMainImg.jpg"
            alt="Category Section Main Imgage"
          />
        </div>
        <aside
          className={`${showAsideFilter
            ? "asideFilter-wrapper--show"
            : "asideFilter-wrapper"
            }`}
        >
          <AsideFilter
            allCollectionArray={allCollectionArray}
            filterRequest={filterRequest}
          />
        </aside>
        <div className="filter-wrapper">
          <CategoryFilter onClickFunc={callAsideFilter} setResult={array} />
        </div>
        <div
          style={{
            backgroundColor: "rgba(100, 85, 45, 0.5)",
            width: "850px",
            height: "510px",
          }}
          className="categoryCards-wrapper"
        >
          {!Array.isArray(filtredArray)
            ? allCollectionArray.map((item, i) => {
              const { name, insertNumber, alt, bestseller, newProduct } = item;
              return (
                <p key={i}>
                  {i} Product:{name} insertNumber:{insertNumber} alt:{alt} bestseller:
                  {bestseller} newProduct:{newProduct}
                </p>
              );
            })
            : filtredArray.map((item, i) => {
              const { name, statusProduct, insertNumber, collectionName, categories, metal } = item;
              return (
                <p key={i}>
                  {i} name:{name} statusProduct:{statusProduct} categories:{categories}
                </p>
              );
            })}
        </div>
        <div
          style={{ backgroundColor: "grey", width: "396px", height: "88px" }}
          className="paginnation-wrapper"
        >
          <Items items={currentItem} loading={loading} />
          <Pagination itemsPerPage={itemsPerPage} totalItems={items.length} />
        </div>
      </div>
    </div>
  );
};

CatalogSectionPage.defaultProps = {
  alreadyFilteredArray: [],
};

export default CatalogSectionPage;
