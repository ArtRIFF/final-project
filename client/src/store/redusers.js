import { createReducer } from "@reduxjs/toolkit";
import * as actions from "./actions";

export const defaultState = {
  modalRender: false,
  modalDeleteRender: false,
  chosenCard: null,
  productArray: [],
  cartArray: [],
  favoriteProd: [],
  newCollectionProduct: [],
  category: {
    earrings: [],
  },
  bestsellers: [],
  outlet: [],
};

export default createReducer(defaultState, {
  [actions.setModalRender]: (state, { payload }) => {
    state.modalRender = payload;
  },
  [actions.setNewCollectionProduct]: (state, { payload }) => {
    state.newCollectionProduct = payload;
  },
  [actions.setCategoryEarrings]: (state, { payload }) => {
    state.category.earrings = payload;
  },
  [actions.setBestsellers]: (state, {payload}) => {
    state.bestsellers = payload;
  },
  [actions.setOutlet]: (state, {payload}) => {
    state.outlet = payload;
  },
  // [actions.toDefault]: () => defaultState,
});
