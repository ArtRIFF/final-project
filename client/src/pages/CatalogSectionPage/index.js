import CatalogSectionPage from "./CatalogSectionPage";
export default CatalogSectionPage;
