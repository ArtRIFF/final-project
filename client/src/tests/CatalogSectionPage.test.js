import CatalogSectionPage from "../pages/CatalogSectionPage/CatalogSectionPage";
import "@testing-library/jest-dom";
import { render, cleanup, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import store from "../store/store";

import LoadingSpinner from "../pages/CatalogSectionPage/components/LoadingSpinner/LoadingSpinner";


afterEach(cleanup);

it("renders with snapshot", () => {
  const catalogSectionPage = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <CatalogSectionPage />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(catalogSectionPage).toMatchSnapshot();
});

describe("CatalogSectionPage works correctly", () => {
  it("CatalogSectionPage renders", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <CatalogSectionPage />
        </BrowserRouter>
      </Provider>
    );
    expect(screen.getByRole('img')).toBeInTheDocument();
  });

 
});
