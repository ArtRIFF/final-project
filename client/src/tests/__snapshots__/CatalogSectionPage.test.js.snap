// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with snapshot 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <div
    className="breadcrumbs__catalog"
  >
    <div
      className="breadcrumb container"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        <span>
          Home
        </span>
         
         / 
      </a>
    </div>
  </div>
  <div
    className="grid-wrapper"
  >
    <div
      className="catalogPageImg-wrapper"
    >
      <img
        alt="Category Section Main Imgage"
        src="/img/catalogSectionPage/CategorySectionMainImg.jpg"
      />
    </div>
    <aside
      className="asideFilter-wrapper"
    >
      <div
        className="aside-filter"
      >
        <div
          className="aside-filter__header"
        >
          <h3
            className="aside-filter__title"
          >
            Filters
          </h3>
          <button
            className="aside-filter__button"
            onClick={[Function]}
          >
            Reset all
          </button>
        </div>
        <div
          className="aside-filter__wrapper"
        >
          <div
            className="filter-parameter filter-parameter__hide"
          >
            <div
              className="filter-parameter__header"
            >
              <h5
                className="filter-parameter__title"
              >
                Price
              </h5>
              <button
                className="filter-parameter__button"
                onClick={[Function]}
              />
            </div>
            <div
              className="filter-parameter__container price-container"
            >
              <input
                data-index={0}
                onChange={[Function]}
                placeholder={0}
                type="text"
                value={0}
              />
              <span>
                -
              </span>
              <input
                data-index={1}
                onChange={[Function]}
                placeholder={0}
                type="text"
                value={0}
              />
            </div>
          </div>
          <div
            className="filter-parameter filter-parameter__hide"
          >
            <div
              className="filter-parameter__header"
            >
              <h5
                className="filter-parameter__title"
              >
                Collection
              </h5>
              <button
                className="filter-parameter__button"
                onClick={[Function]}
              />
            </div>
            <div
              className="filter-parameter__container"
            >
              <label
                className="filter-parameter__checkbox"
                data-category-name="vintage"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Vintage
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="seabed"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Seabed
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="refinement"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Refinement
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="capsule"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Capsule
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="max spass"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Max Spass
                </span>
              </label>
            </div>
          </div>
          <div
            className="filter-parameter filter-parameter__hide"
          >
            <div
              className="filter-parameter__header"
            >
              <h5
                className="filter-parameter__title"
              >
                Insert
              </h5>
              <button
                className="filter-parameter__button"
                onClick={[Function]}
              />
            </div>
            <div
              className="filter-parameter__container"
            >
              <label
                className="filter-parameter__checkbox"
                data-category-name="Without inserting"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Without insertion
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="pearl"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Pearl
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="zirconia"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Zirconia
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="diamond"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Diamond
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="Pearl and diamonds"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Pearl and diamonds
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="semi-precious"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Semi-precious
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="nacre"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Nacre
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="amethyst"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Amethyst
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="onixl"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Onixl
                </span>
              </label>
            </div>
          </div>
          <div
            className="filter-parameter filter-parameter__hide"
          >
            <div
              className="filter-parameter__header"
            >
              <h5
                className="filter-parameter__title"
              >
                Metal
              </h5>
              <button
                className="filter-parameter__button"
                onClick={[Function]}
              />
            </div>
            <div
              className="filter-parameter__container"
            >
              <label
                className="filter-parameter__checkbox"
                data-category-name="Gold"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Gold
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="Silver"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Silver
                </span>
              </label>
            </div>
          </div>
          <div
            className="filter-parameter filter-parameter__hide"
          >
            <div
              className="filter-parameter__header"
            >
              <h5
                className="filter-parameter__title"
              >
                Metal color
              </h5>
              <button
                className="filter-parameter__button"
                onClick={[Function]}
              />
            </div>
            <div
              className="filter-parameter__container"
            >
              <label
                className="filter-parameter__checkbox"
                data-category-name="White"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  White
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="Yellow"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Yellow
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="Red"
                onClick={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Red
                </span>
              </label>
            </div>
          </div>
          <div
            className="filter-parameter filter-parameter__hide"
          >
            <div
              className="filter-parameter__header"
            >
              <h5
                className="filter-parameter__title"
              >
                Product Type
              </h5>
              <button
                className="filter-parameter__button"
                onClick={[Function]}
              />
            </div>
            <div
              className="filter-parameter__container"
            >
              <label
                className="filter-parameter__checkbox"
                data-category-name="bracelet"
                onChange={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Bracelet
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="ring wedding-ring engagement-ring"
                onChange={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Ring
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="earring child-earrings"
                onChange={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Earring
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="cross"
                onChange={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Cross
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="pendant"
                onChange={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Pendant
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="pearl"
                onChange={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Pearl
                </span>
              </label>
            </div>
          </div>
          <div
            className="filter-parameter filter-parameter__hide"
          >
            <div
              className="filter-parameter__header"
            >
              <h5
                className="filter-parameter__title"
              >
                Product Status
              </h5>
              <button
                className="filter-parameter__button"
                onClick={[Function]}
              />
            </div>
            <div
              className="filter-parameter__container"
            >
              <label
                className="filter-parameter__checkbox"
                data-category-name="BESTSELLER"
                onChange={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Bestseller
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="OUTLET"
                onChange={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  Outlet
                </span>
              </label>
              <label
                className="filter-parameter__checkbox"
                data-category-name="NEW"
                onChange={[Function]}
              >
                <input
                  type="checkbox"
                />
                <span>
                  New Collection
                </span>
              </label>
            </div>
          </div>
        </div>
      </div>
    </aside>
    <div
      className="filter-wrapper"
    >
      <div
        className="category-filter"
      >
        <div
          className="category-filter--btn"
          onClick={[Function]}
        >
          <div>
            <button
              className="section__btn-subscribe"
            >
              Filter
            </button>
          </div>
        </div>
        <div
          className="category-filter__result"
        >
          Results: 
          <span />
        </div>
        <select
          data-testid="select-sortBy"
          defaultValue="DEFAULT"
          name="select-sortBy"
          onChange={[Function]}
        >
          <option
            disabled={true}
            value="DEFAULT"
          >
            Sort by
          </option>
          <option
            value="currentPrice"
          >
            Price
          </option>
          <option
            value="date"
          >
            Newest
          </option>
        </select>
      </div>
    </div>
    <div
      className="categoryCards-wrapper"
    >
      <svg
        height="110px"
        preserveAspectRatio="xMidYMid"
        style={
          Object {
            "display": "block",
            "margin": "auto",
            "shapRendering": "auto",
          }
        }
        viewBox="0 0 100 100"
        width="110px"
      >
        <g
          transform="rotate(0 50 50)"
        >
          <rect
            fill="#0a0a0a"
            height="12"
            rx="4"
            ry="4.92"
            width="8"
            x="46"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.9166666666666666s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(30 50 50)"
        >
          <rect
            fill="#0a0a0a"
            height="12"
            rx="4"
            ry="4.92"
            width="8"
            x="46"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.8333333333333334s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(60 50 50)"
        >
          <rect
            fill="#0a0a0a"
            height="12"
            rx="4"
            ry="4.92"
            width="8"
            x="46"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.75s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(90 50 50)"
        >
          <rect
            fill="#0a0a0a"
            height="12"
            rx="4"
            ry="4.92"
            width="8"
            x="46"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.6666666666666666s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(120 50 50)"
        >
          <rect
            fill="#0a0a0a"
            height="12"
            rx="4"
            ry="4.92"
            width="8"
            x="46"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.5833333333333334s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(150 50 50)"
        >
          <rect
            fill="#0a0a0a"
            height="12"
            rx="4"
            ry="4.92"
            width="8"
            x="46"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.5s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(180 50 50)"
        >
          <rect
            fill="#0a0a0a"
            height="12"
            rx="4"
            ry="4.92"
            width="8"
            x="46"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.4166666666666667s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(210 50 50)"
        >
          <rect
            fill="#0a0a0a"
            height="12"
            rx="4"
            ry="4.92"
            width="8"
            x="46"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.3333333333333333s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(240 50 50)"
        >
          <rect
            fill="#0a0a0a"
            height="12"
            rx="4"
            ry="4.92"
            width="8"
            x="46"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.25s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(270 50 50)"
        >
          <rect
            fill="#0a0a0a"
            height="12"
            rx="4"
            ry="4.92"
            width="8"
            x="46"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.16666666666666666s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(300 50 50)"
        >
          <rect
            fill="#0a0a0a"
            height="12"
            rx="4"
            ry="4.92"
            width="8"
            x="46"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.08333333333333333s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(330 50 50)"
        >
          <rect
            fill="#0a0a0a"
            height="12"
            rx="4"
            ry="4.92"
            width="8"
            x="46"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="0s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
      </svg>
    </div>
    <div
      className="paginnation-wrapper"
    >
      <div
        style={
          Object {
            "marginTop": "30px",
          }
        }
      >
        <ul
          className="pagination-container"
        >
          <span
            className="page-button disabled"
            onClick={[Function]}
          >
            Prev
          </span>
          <span
            className="page-button "
            onClick={[Function]}
          >
            Next
          </span>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
